compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -644,7 +631,7 @@ impl<'context> Elaborator<'context> {
         let has_no_predicates_attribute = func.attributes().is_no_predicates();
         let should_fold = func.attributes().is_foldable();
         let has_inline_attribute = has_no_predicates_attribute || should_fold;
-        let is_pub_allowed = self.pub_allowed(func);
+        let is_pub_allowed = self.pub_allowed(func, in_contract);
```

Comment:
Nit: looks like we can move this up and use it in the `unnecessary_pub_return` lint